lib/cacheable/response_cache_handler.rb

```diff
@@ -149,10 +148,9 @@ def page_too_old(timestamp, cache_age_tolerance)
       !timestamp || timestamp < (Time.now.to_i - cache_age_tolerance)
     end
 
-    def cache_return!(message, &block)
+    def cache_return!(message, response)
```

Comment:
It’s not clear to me that this is still useful when the rails rendering calls are moved to the caller of ResponseCacheHandler.

Reply:
You mean the method itself?

Reply:
The method and throw-based control flow

Reply:
I think I prefer it to the alternative of checking some conditions after each attempt to serve from the cache and then returning early.

```ruby
serve_from_browser_cache(versioned_key_hash)
return @repsonse if cache_return?

# or

return @repsonse if serve_from_browser_cache(versioned_key_hash)
```

Here's some benchmarks

**throw-catch**
```
.Warming up --------------------------------------
          cache_miss     2.467k i/100ms
Calculating -------------------------------------
          cache_miss     25.670k (± 3.4%) i/s -    128.284k in   5.003225s
```

**return early**
```
.Warming up --------------------------------------
          cache_miss     3.257k i/100ms
Calculating -------------------------------------
          cache_miss     34.456k (± 3.5%) i/s -    172.621k in   5.015960s
```

Reply:
After fixing it up and seeing the benchmarks I've decided to remove the throw-catch logic.